dolphinscheduler-api/src/main/java/org/apache/dolphinscheduler/api/service/DataSourceService.java

```diff
@@ -161,15 +159,15 @@
             return result;
         }
         //check password，if the password is not updated, set to the old password.
-        JSONObject paramObject = JSON.parseObject(parameter);
-        String password = paramObject.getString(Constants.PASSWORD);
+        ObjectNode paramObject = JSONUtils.parseObject(parameter);
+        String password = paramObject.path(Constants.PASSWORD).asText();
         if (StringUtils.isBlank(password)) {
             String oldConnectionParams = dataSource.getConnectionParams();
-            JSONObject oldParams = JSON.parseObject(oldConnectionParams);
-            paramObject.put(Constants.PASSWORD, oldParams.getString(Constants.PASSWORD));
+            ObjectNode oldParams = JSONUtils.parseObject(oldConnectionParams);
+            paramObject.put(Constants.PASSWORD, oldParams.path(Constants.PASSWORD).asText());
         }
         // connectionParams json
-        String connectionParams = paramObject.toJSONString();
+        String connectionParams = JSONUtils.toJsonString(paramObject);
```

Comment:
paramObject is ObjectNode ,why not use toString directly ?

Reply:
ok，I will fix all similar problems